test/test_petab_import.py

```diff
@@ -68,7 +69,47 @@ def test_3_optimize(self):
                 result.optimize_result.get_for_key('fval')[0]))
 
 
+class SpecialFeaturesTest(unittest.TestCase):
+
+    def test_replicates(self):
+        """
+        Use a model that has replicates and check that all data points are
+        inserted.
+        """
+        # import a model with replicates at some time points and observables
+        importer = pypesto.PetabImporter.from_folder(
+            folder_base + "Schwen_PONE2014")
+
+        # create amici.ExpData list
+        edatas = importer.create_edatas()
+
+        # convert to dataframes
+        amici_df = amici.getDataObservablesAsDataFrame(
+            importer.create_model(), edatas)
+        # reduce to data columns
+        amici_df = amici_df[[col for col in amici_df.columns
+                             if col.startswith("observable_")
+                             and not col.endswith("_std")]]
+
+        meas_df = importer.petab_problem.measurement_df
+
+        # amici_df subset measurement_df
+        for _, row in meas_df.iterrows():
+            val = row.measurement
+            # test if amici_df contains this value somewhere
+            # this will test up to np.isclose closeness
+            self.assertTrue(np.isclose(amici_df, val).any(axis=1).any())
```

Comment:
This will not detect a lost data point if there is another one with the same value, will it?

Could flatten and sort the array and check if each entry is close to corresponding the sorted measurement column for that condition.

Reply:
No, it will not detect such a point. But the points in the Schwen model are quite diverse such that we definitely have many cases where both values need to end up in the file.

I was not sure how stable the amici dataframe format is. If one can assume that the time column will always be denoted ``time`` and the observable columns ``observable_{OBS_ID}`` (hej, there the observable_ is again ;) ), then I could indeed do that.

Reply:
funnily, in the measurement table, the lines https://github.com/LoosC/Benchmark-Models/blob/hackathon/hackathon_contributions_new_data_format/Schwen_PONE2014/measurementData_Schwen_PONE2014.tsv#L27 and following are exactly the same though (don't know if that makes sense).

Reply:
Very reproducible measurement, yeah, wow.

Reply:
> Could flatten and sort the array and check if each entry is close to corresponding the sorted measurement column for that condition.

Done that now. The only thing that could be missed is when the wrong observable gets assigned the data.

Reply:
now recovering the full details, including which data for which observable id (and time point) :)

Reply:
Great